### PR TITLE
Move vagrant activation to a parameter

### LINKED ISSUE #12611: Automate the packaging tests
Right now there are some packaging tests that run as [bats](https://github.com/sstephenson/bats) scripts. They are destructive to the system they run on and only work on certain operating systems. It doesn't make sense to test the RPM build on Debian, nor does it make sense to test the DEB build on CentOS.

I propose we automate some testing of these using vagrant or a similar technology. Vagrant sounds like a good place to start at least.

### PR DESCRIPTION
**Pull request by nik9000**
Closes #12611

### CODE REVIEW

**Comment by nik9000** [comment]
This moves the vagrant profile activation to a parameter like @dadoonet  and @rjernst asked for in #12646.

File: qa/pom.xml [hunk 1]
```diff
@@ -152,6 +152,11 @@
     <profiles>
         <profile>
             <id>vagrant</id>
+            <activation>
+                <property>
+                    <name>vagrant</name>
```
Comment (dadoonet) [inline qa/pom.xml]: Robert [proposed](https://github.com/elastic/elasticsearch/pull/12646#issuecomment-129041593) to use `tests.vagrant` which is more compliant to other properties we have.

Comment (jpountz) [inline qa/pom.xml]: +1

Comment (nik9000) [inline qa/pom.xml]: Sure sure. Better idea. I'll do it when I get proper internet.

**Comment by nik9000** [comment]
@dadoonet, renamed the parameter.

**Comment by dadoonet** [comment]
It looks good to me.

**Comment by nik9000** [comment]
Squashed and merged.

**Pull request merged**